### PR TITLE
Enable editor path to be sent in related code request

### PR DESCRIPTION
**Pull request by edzkite**
Unblocks https://github.com/kiteco/kiteco/issues/12103 for Atom and VSCode

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -236,14 +236,15 @@ const KiteAPI = {
     );
   },
 
-  requestRelatedCode(editor, filename, line, selection) {
```
Comment (tonycheang) [inline lib/index.js:239]: minor nit: consider grouping editor and editor_path?

**Review by tonycheang — APPROVED** [review]

File: lib/index.js [hunk 2]
```diff
@@ -236,14 +236,15 @@ const KiteAPI = {
...
+  requestRelatedCode(editor, filename, line, selection, editor_path) {
     return this.request(
       {
         path: '/codenav/editor/related',
         method: 'POST',
       },
       JSON.stringify({
         editor,
+        editor_path,
```
Comment (tonycheang) [inline lib/index.js]: IntelliJ is currently sending `editor_install_path` to kited. We can either change that in IntelliJ to `editor_path` too or go with `editor_install_path` here. I slightly prefer the verbose `editor_install_path`, but what do you think?

Comment (edzkite) [reply lib/index.js]: Ah, thanks for letting me know. I'll change this to `editor_install_path`.

File: lib/index.js (hunk 1 above)
Comment (edzkite) [reply lib/index.js:239]: I prefer to keep it separate here, since it would be a non-breaking change, both from the editor and Kited side.

Comment (tonycheang) [reply lib/index.js:239]: Well, when we add selection, the function signature would be `requestRelatedCode(editor, filename, line, editor_install_path, selection)` or something. It seems slightly better to group `(editor, editor_install_path, filename, line)` while we don't officially support code nav. I don't think anything would have to change about the json sent to kited. But it's not a big deal either way

Comment (edzkite) [reply lib/index.js:239]: Given that we hard lock Kite API to each editor plugin, I'm okay with making this change. 

Generally, I prefer to add new params to the end of a released API; that way I can release the change and use the new version in one consumer of the API without needing to change how it's called in other consumers until desired.

**Review by tonycheang — APPROVED** [review]

**Pull request merged**